src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/WindowsRuntime/ExceptionSupport.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+namespace Internal.Runtime.InteropServices.WindowsRuntime
+{
+    public static class ExceptionSupport
+    {
+        /// <summary>
+        /// Attach restricted error information to the exception if it may apply to that exception, returning
+        /// back the input value
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Exception AttachRestrictedErrorInfo(this Exception e)
```

Comment:
I do not think this should be extension method.

Reply:
CoreRT has same method with extension method.  Maybe I should change both to normal method instead of extension method.

Reply:
Yes, that would be nice.

Reply:
Add a TODO for myself.